portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -19,14 +19,73 @@
 import javax.servlet.http.HttpServlet;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
+import com.google.gson.Gson;
+import java.util.ArrayList;
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.Query.SortDirection;
+import java.util.Collections;
 
 /** Servlet that returns some example content. TODO: modify this file to handle comments data */
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
-
+  ArrayList <String> comments = new ArrayList <String> ();
+  ArrayList <String> comments_fin = new ArrayList <String> ();
+    
+    
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    
     response.setContentType("text/html;");
-    response.getWriter().println("<h1>Hello world!</h1>");
+    int i = 0;
+    int l = comments_fin.size();
+    while(i<l){
+      if(comments_fin.get(i)==null){
+        comments_fin.remove(i);
+        l--;
+      }
+      else{
+        i++;
+      }
+
+    }
+    
+    response.getWriter().println(comments_fin);
+  }
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    String comment = request.getParameter("comment");
+    comments_fin.clear();
+    comments.clear();
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    
+    int curr_num_comments=0;
+    
+    int max_num_comments;
+    try {
+      max_num_comments = Integer.parseInt(request.getParameter("num_comment"));
+    } catch (NumberFormatException e) {
+      System.err.println("Invalid Number, Use Default: 3");
+      max_num_comments = 3;
+    }
+   
+    Entity taskEntity = new Entity("Feedback");
+    taskEntity.setProperty("comment",comment);
+    datastore.put(taskEntity);
+    Query query = new Query("Feedback");
+    PreparedQuery results = datastore.prepare(query);
+    for (Entity entity : results.asIterable()) {
+      comments.add((String)entity.getProperty("comment"));
+      System.out.println("Take2"+ (String)entity.getProperty("comment"));
+    }
+    Collections.shuffle(comments);
+    while(curr_num_comments<max_num_comments && curr_num_comments<comments.size()){
```

Comment:
This is a super small thing, but could you make the following adjustments on spacing? 

- operands should always have a space between themselves and the variables they operate over (like the && here, and not like the <)
- code blocks (conditionals and loops) should have spaces between the keyword (while for, etc), the control (components in parantheses) and the opening bracket (more like line 79 and not like line 84)